src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiListType.java

```diff
@@ -27,26 +26,30 @@ private ApiListType(
     super(ApiTypeName.LIST, valueType, DataTypes.listOf(valueType.cqlType(), isFrozen), apiSupport);
   }
 
+  @Override
+  public boolean isFrozen() {
+    return cqlType.isFrozen();
+  }
+
   @Override
   public ColumnDesc columnDesc() {
     return new ListColumnDesc(valueType.columnDesc(), ApiSupportDesc.from(this));
   }
 
+  /**
+   * Creates a new {@link ApiListType} from the given ApiDataType value.
+   *
+   * <p>ApiSupport for valueType should be already validated.
+   */
   static ApiListType from(ApiDataType valueType, boolean isFrozen) {
     Objects.requireNonNull(valueType, "valueType must not be null");
-
-    if (isValueTypeSupported(valueType)) {
-      return new ApiListType(
-          (PrimitiveApiDataTypeDef) valueType, defaultApiSupport(isFrozen), isFrozen);
-    }
-
-    throw new IllegalArgumentException(
-        "valueType must be primitive type, valueType%s".formatted(valueType));
+    return new ApiListType(
```

Comment:
-1 for a non private function assuming validation has been done, please put the isValueTypeSupported() check back